### PR TITLE
Addition of role icons

### LINKED ISSUE #191: Role icons
### Summary

implement the ability to view and edit role icons

### What is the feature request for?

The core library

### The Problem

Role icons are a ✨ _sparkly new feature_ ✨ that i'd like to be able to manage with a bot.

### The Ideal Solution

Implement them per the API specs. thats it.

https://discord.com/developers/docs/topics/permissions#role-object

https://discord.com/developers/docs/resources/guild#modify-guild-role

### The Current Solution

It might be possible to use the internal http features of the library to do it 

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by veni-vidi-code**
## Summary

Adds new role icons to the libary

Closes #191 

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by HexCodeFFF** [comment]
I own a level 3 server, I'll give this a test in a bit.

**Review by parker02311 — APPROVED** [review]
lgtm

File: nextcord/guild.py [hunk 1]
```diff
@@ -78,6 +78,7 @@
 from .file import File
 
 
+
```
Comment (ooliver1) [inline nextcord/guild.py:81]: hm whats this for

File: nextcord/guild.py [hunk 2]
```diff
@@ -2475,6 +2479,8 @@ async def create_role(
             Defaults to ``False``.
         reason: Optional[:class:`str`]
             The reason for creating this role. Shows up on the audit log.
+        icon: Optional[Union[:class:`str`, :class:`bytes`]
```
Comment (ooliver1) [inline nextcord/guild.py:2482]: forgot to close the ``Optional[]``

**Review by ooliver1 — CHANGES_REQUESTED** [review]

**Comment by Skelmis** [comment]
@veni-vidi-code any updates on this?

File: nextcord/guild.py (hunk 2 above)
Comment (veni-vidi-code) [inline nextcord/guild.py:2482]:
```suggestion
        icon: Optional[Union[:class:`str`, :class:`bytes`]]
```

File: nextcord/guild.py (hunk 1 above)
Comment (veni-vidi-code) [inline nextcord/guild.py:81]:
```suggestion
```

**Review by vcokltfre — APPROVED** [review]

**Pull request merged**